hips/hip-0019.md

```diff
@@ -0,0 +1,120 @@
+---
+hip: "0019"
+title: "Adoption of CUE for Values Processing and Validation in Helm"
+authors: ["Jiří Altman <jiri.altman@usu.com>", "Jan von Löwenstein <jan.von.loewenstein@sap.com>"]
+created: "2024-08-07"
+type: "feature"
+status: "draft"
+---
+
+## Abstract
+
+This HIP proposes the integration of CUE (Configuration Unification Engine) as the primary tool for processing and validating Helm values files, replacing the existing JSON Schema support. As a superset of YAML, JSON, and JSON Schema, CUE extends existing capabilities while maintaining complete backward compatibility. This ensures that all current Helm charts and configurations remain fully supported, with the added benefits of enhanced validation, improved defaulting mechanisms, and easier schema management.
```

Comment:
Also, I always thought CUE stood for "Configure Unify Execute" (and not: "Configuration Unification Engine"). Where does this alternate come from?